### PR TITLE
graphics: fix ephemeral spawning bug

### PR DESCRIPTION
**Pull request by kanyesthaker**
## Description of Changes
Fixes an ephemeral bug where the player would appear below the surface of the bar for 1 frame before the physics ticker begins

The issue is that the player's physics are paused before the player fully loads. When they're paused the player has a position of (0, 0, 0) (the default). The camera only snaps to the player's position once the physics ticker starts ticking, which is delayed by a few frames. By syncing the saved vector, the player's position, and the camera position on load, we prevent any desync between the three.

## Testing Checklists
If your change adds critical functionality that should be tested on every PR, edit ``pull_requeste_template.md`` and add it to the checklist.
### Basic Author Checklist 
The PR author should verify that the following are fully functional.
- [x] With 1 user:
  - [x] The world loads, the user can move/look around, and physics behave properly
  - [x] User can pause their client, mute/unmute themselves, and change their name
- [x] With 2 users:
  - [x] A second user can join the same room, and both users can see each others model/video and hear each others audio
  - [x] State changes (name changes, A/V toggles) propagate between peers as expected

### PR Specific Author Checklist
What did you do to test your change in particular? For example, if your changes touches A/V functionality, how did you stress-test this part of the product?
- [x] Refreshed 10 times to ensure no flickering, with multiple peers

### Advanced Author Checklist
These are more advanced checks that might be helpful for some PRs. Completing these is at the discretion of the author and reviewer.
- [ ] With 2 users, memory stays within a reasonable interval over a 5 minute period
- [ ] With 2 users, the server remains healthy after an extended amount of time in the world
- [ ] With 3+ users, the server and client behave in a manner consistent with 2 peers

### Basic Reviewer Checklist
The reviewer of this PR should load SpaceBar off this branch and ensure the following work as expected:
- [ ] 2 user can join the world and see each other's model/video, and hear each other's audio
- [ ] Movement and collisions behave as expected
- [ ] Changing some state (name, microphone, video) for one user propagates correctly to the other user

### CODE REVIEW

File: src/scene/players/UserPlayerEntity.tsx [hunk 1]
```diff
@@ -69,16 +69,22 @@ UserPlayerEntityProps) => {
...
     // ! Kanyes: This is a hack until we have a dedicated spawn area
     const minX = -10; // furthest left the player will spawn
     const maxX = 10; // furthest right the player will spawn
     const xDisplacement = Math.floor(Math.random() * (maxX - minX + 1)) + minX;
-    playerCapsuleRef.current.position.copy(
-      new Vector3(...PlayerConstants.spawnPosition).add(
-        new Vector3(xDisplacement, 0, 0),
-      ),
+    let spawnPosition = new Vector3(...PlayerConstants.spawnPosition).add(
+      new Vector3(xDisplacement, 0, 0),
```
Comment (shalins) [inline src/scene/players/UserPlayerEntity.tsx:83]: to make future reviews faster, could you perhaps highlight where the actual bug was fixed? where was the issue? what changed? helps build context for reviewers so we can catch it in future reviews :)

in this case, im assuming that the bug was caused by not setting the `savedPositionRef` -- is that correct? why did that cause the player to spawn below the surface?

Comment (shalins) [inline src/scene/players/UserPlayerEntity.tsx:83]: this is out of curiosity, why are we setting just the `xDisplacement`?

**Review by shalins — APPROVED** [review]

File: src/scene/players/UserPlayerEntity.tsx (hunk 1 above)
Comment (kanyesthaker) [reply src/scene/players/UserPlayerEntity.tsx:83]: this is a hack until we get a good spawn area -- the little rug in front of the bar extends along the x-axis.

Comment (kanyesthaker) [reply src/scene/players/UserPlayerEntity.tsx:83]: my bad -- The issue is that the player's physics are paused before the player fully loads. When they're paused the player has a position of (0, 0, 0) (the default). The camera only snaps to the player's position once the physics ticker starts ticking, which is delayed by a few frames. By syncing the saved vector, the player's position, and the camera position on load, we prevent any desync between the three. Updated PR description

Comment (shalins) [reply src/scene/players/UserPlayerEntity.tsx:83]: great, thanks for clarifying and for pushing the fix!

**Pull request merged**